sonic-xcvrd/xcvrd/xcvrd_utilities/port_mapping.py

```diff
@@ -173,42 +189,54 @@ def handle_port_update_event(sel, asic_context, stop_event, logger, port_change_
                 fvp['op'] = op
                 fvp['FILTER'] = port_tbl.filter
                 # Soak duplicate events and consider only the last event
-                port_event_cache[key+port_tbl.db_name+port_tbl.table_name] = fvp
+                port_event_cache[(key, port_tbl.db_name, port_tbl.table_name)] = fvp
 
         # Now apply filter over soaked events
         for key, fvp in port_event_cache.items():
+            port_name = key[0]
+            db_name = key[1]
+            table_name = key[2]
             port_index = int(fvp['index'])
             port_change_event = None
-            diff = {}
             filter = fvp['FILTER']
             del fvp['FILTER']
             apply_filter_to_fvp(filter, fvp)
 
-            if key in PortChangeEvent.PORT_EVENT:
-               diff = dict(set(fvp.items()) - set(PortChangeEvent.PORT_EVENT[key].items()))
-               # Ignore duplicate events
-               if not diff:
-                  PortChangeEvent.PORT_EVENT[key] = fvp
-                  continue
-            PortChangeEvent.PORT_EVENT[key] = fvp
+            if key in PortChangeEvent.thread_local_data.PORT_EVENT:
+                # Compare current event with last event on this key, to see if
+                # there's really a change on the contents.
+                diff = set(fvp.items()) - set(PortChangeEvent.thread_local_data.PORT_EVENT[key].items())
+                # Ignore duplicate events
+                if not diff:
+                   PortChangeEvent.thread_local_data.PORT_EVENT[key] = fvp
+                   continue
+            # Update the latest event to the cache
+            PortChangeEvent.thread_local_data.PORT_EVENT[key] = fvp
```

Comment:
@longhuan-cisco separate PR for this should work for CMIS mgr as well, so I don't see any reason it cannot be a separate PR. It will less of a work for release mgr to revert in case of smaller change to be reverted that a feature + enhancement/fix

Reply:
sure, created separate PR https://github.com/sonic-net/sonic-platform-daemons/pull/430